packages/api/source/controllers/RoleControllers.ts

```diff
@@ -0,0 +1,133 @@
+import type { Document } from "mongoose";
+import joi from "joi";
+import type { Role, RolePage } from "../types";
+
+import RoleModel from "../models/RoleModel";
+
+import {
+    constants,
+    BadRequestError,
+    NotFoundError,
+    extractIds,
+} from "../utils";
+
+const createSchema = joi.object({
+    name: joi.string().max(512).required(),
+    priviledges: joi.array().items(joi.string().max(512).required()),
+});
+
+const updateSchema = joi.object({
+    name: joi.string(),
+    priviledges: joi.array().items(joi.string().max(512)),
+});
```

Comment:
Typo isn't corrected. We spoke about this.

Reply:
Done